cmd/register/files.go

```diff
@@ -0,0 +1,259 @@
+				Name:         workflowSpec.Template.Id.Name,
+				Version:      filesConfig.version,
+			},
+			Spec: workflowSpec,
+		})
+		return err
+	case *admin.TaskSpec:
+		taskSpec := message.(*admin.TaskSpec)
+		_, err := cmdCtx.AdminClient().CreateTask(ctx, &admin.TaskCreateRequest{
+			Id: &core.Identifier{
+				ResourceType: core.ResourceType_TASK,
+				Project:      config.GetConfig().Project,
+				Domain:       config.GetConfig().Domain,
+				Name:         taskSpec.Template.Id.Name,
+				Version:      filesConfig.version,
+			},
+			Spec: taskSpec,
+		})
+		return err
+	default:
+		return errors.New(fmt.Sprintf("Failed registering unknown entity  %v", message))
+	}
+}
+
+func hydrateNode(node *core.Node) error {
+	targetNode := node.Target
+	switch targetNode.(type) {
+	case *core.Node_TaskNode:
+		taskNodeWrapper := targetNode.(*core.Node_TaskNode)
+		taskNodeReference := taskNodeWrapper.TaskNode.Reference.(*core.TaskNode_ReferenceId)
+		hydrateIdentifier(taskNodeReference.ReferenceId)
+	case *core.Node_WorkflowNode:
+		workflowNodeWrapper := targetNode.(*core.Node_WorkflowNode)
+		switch workflowNodeWrapper.WorkflowNode.Reference.(type) {
+		case *core.WorkflowNode_SubWorkflowRef:
+			subWorkflowNodeReference := workflowNodeWrapper.WorkflowNode.Reference.(*core.WorkflowNode_SubWorkflowRef)
+			hydrateIdentifier(subWorkflowNodeReference.SubWorkflowRef)
+		case *core.WorkflowNode_LaunchplanRef:
+			launchPlanNodeReference := workflowNodeWrapper.WorkflowNode.Reference.(*core.WorkflowNode_LaunchplanRef)
+			hydrateIdentifier(launchPlanNodeReference.LaunchplanRef)
+		default:
+			errors.New(fmt.Sprintf("Unknown type %T", workflowNodeWrapper.WorkflowNode.Reference))
+		}
+	case *core.Node_BranchNode:
+		branchNodeWrapper := targetNode.(*core.Node_BranchNode)
+		hydrateNode(branchNodeWrapper.BranchNode.IfElse.Case.ThenNode)
+		if len(branchNodeWrapper.BranchNode.IfElse.Other) > 0 {
+			for _, ifBlock := range branchNodeWrapper.BranchNode.IfElse.Other {
+				hydrateNode(ifBlock.ThenNode)
+			}
+		}
+		switch branchNodeWrapper.BranchNode.IfElse.Default.(type) {
+		case *core.IfElseBlock_ElseNode:
+			elseNodeReference := branchNodeWrapper.BranchNode.IfElse.Default.(*core.IfElseBlock_ElseNode)
+			hydrateNode(elseNodeReference.ElseNode)
+		case *core.IfElseBlock_Error:
+			// Do nothing.
+		default:
+			return errors.New(fmt.Sprintf("Unknown type %T", branchNodeWrapper.BranchNode.IfElse.Default))
+		}
+	default:
+		return errors.New(fmt.Sprintf("Unknown type %T", targetNode))
+	}
+	return nil
+}
+
+func hydrateIdentifier(identifier *core.Identifier) {
+	if len(identifier.Project) == 0 || identifier.Project == registrationProjectPattern {
+		identifier.Project = config.GetConfig().Project
+	}
+	if len(identifier.Domain) == 0 || identifier.Domain == registrationDomainPattern {
+		identifier.Domain = config.GetConfig().Domain
+	}
+	if len(identifier.Version) == 0 || identifier.Version == registrationVersionPattern {
```

Comment:
for remote entities that use latest version, is the version field optional @katrogan ?